CMakeLists.txt

```diff
@@ -0,0 +1,190 @@
+  find_package(Qt6 6.2 COMPONENTS Quick Core Qml)
+endif()
+
+if(NOT Qt6_FOUND)
+    find_package(Qt5 5.12 COMPONENTS Quick Core Qml REQUIRED)
+    set(QT_MAJOR_VERSION Qt5)
+endif()
+
+message(STATUS "Using ${QT_MAJOR_VERSION}")
+
+if(BUILD_SHARED_LIBS)
+  set(QML_PLUGIN_LINK_TYPE SHARED)
+else()
+  set(QML_PLUGIN_LINK_TYPE STATIC)
+endif()
+
+set(QML_BOX2D_SOURCES
+    box2dbody.cpp
+    box2dbody.h
+    box2dcontact.cpp
+    box2dcontact.h
+    box2ddebugdraw.cpp
+    box2ddebugdraw.h
+    box2ddistancejoint.cpp
+    box2ddistancejoint.h
+    box2dfixture.cpp
+    box2dfixture.h
+    box2dfrictionjoint.cpp
+    box2dfrictionjoint.h
+    box2dgearjoint.cpp
+    box2dgearjoint.h
+    box2djoint.cpp
+    box2djoint.h
+    box2dmotorjoint.cpp
+    box2dmotorjoint.h
+    box2dmousejoint.cpp
+    box2dmousejoint.h
+    box2dplugin.cpp
+    box2dplugin.h
+    box2dprismaticjoint.cpp
+    box2dprismaticjoint.h
+    box2dpulleyjoint.cpp
+    box2dpulleyjoint.h
+    box2draycast.cpp
+    box2draycast.h
+    box2drevolutejoint.cpp
+    box2drevolutejoint.h
+    box2dropejoint.cpp
+    box2dropejoint.h
+    box2dweldjoint.cpp
+    box2dweldjoint.h
+    box2dwheeljoint.cpp
+    box2dwheeljoint.h
+    box2dworld.cpp
+    box2dworld.h
+    qmldir
+)
+
+add_library(${QML_BOX2D_LIBRARY_NAME} ${QML_PLUGIN_LINK_TYPE}
+  ${QML_BOX2D_SOURCES}
+)
+
+if(ipoSupported)
+  set_property(TARGET ${QML_BOX2D_LIBRARY_NAME} PROPERTY INTERPROCEDURAL_OPTIMIZATION TRUE)
+endif()
+
+set_target_properties(${QML_BOX2D_LIBRARY_NAME} PROPERTIES CXX_VISIBILITY_PRESET hidden)
+
+set(PLUGIN_DEST_DIR "${QMLBOX2D_PLUGIN_BASE_DIR}/Box2D")
+set_target_properties(
+${QML_BOX2D_LIBRARY_NAME}
+PROPERTIES
+    CXX_STANDARD 11
+    CXX_STANDARD_REQUIRED ON
+    CXX_EXTENSIONS NO
+    ARCHIVE_OUTPUT_DIRECTORY "${PLUGIN_DEST_DIR}"
+    LIBRARY_OUTPUT_DIRECTORY "${PLUGIN_DEST_DIR}"
+    RUNTIME_OUTPUT_DIRECTORY "${PLUGIN_DEST_DIR}"
+)
+
+target_include_directories(${QML_BOX2D_LIBRARY_NAME} PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
+
+target_compile_definitions(${QML_BOX2D_LIBRARY_NAME}
+PRIVATE
+    $<$<STREQUAL:${QML_PLUGIN_LINK_TYPE},STATIC>:QT_STATICPLUGIN>
+)
+
+target_link_libraries(${QML_BOX2D_LIBRARY_NAME}
+PRIVATE
+    ${QT_MAJOR_VERSION}::Core
+    ${QT_MAJOR_VERSION}::Quick
+    ${QT_MAJOR_VERSION}::Qml
+PUBLIC
+    Box2D
+)
+
+if(NOT Qt6_FOUND)
+  add_custom_command(TARGET ${QML_BOX2D_LIBRARY_NAME} POST_BUILD
+    COMMAND ${CMAKE_COMMAND} -E copy
+    ${CMAKE_CURRENT_SOURCE_DIR}/qmldir
+    $<TARGET_FILE_DIR:${QML_BOX2D_LIBRARY_NAME}>/qmldir
+  )
+endif()
+
+if(Qt6_FOUND)
+    set_target_properties(${QML_BOX2D_LIBRARY_NAME} PROPERTIES
+        QT_QML_MODULE_VERSION 2.0
+        QT_QML_MODULE_URI     Box2D
+    )
+    qt_add_qml_module(${QML_BOX2D_LIBRARY_NAME}
+      URI Box2D
+      VERSION 2.0
+      PLUGIN_TARGET ${QML_BOX2D_LIBRARY_NAME}
+      OUTPUT_DIRECTORY "${PLUGIN_DEST_DIR}"
+      NO_GENERATE_PLUGIN_SOURCE
+    )
+endif()
+
+if(USE_SYSTEM_BOX2D)
+  find_package(Box2D REQUIRED)
```

Comment:
I've tested on a Ubuntu 18.04 (so with [Box2D 2.3.1](https://packages.ubuntu.com/bionic/libbox2d-dev) installed) and it worked fine.

On my archlinux, I have Box2D 2.4.1 and the package is  named box2d (lowercase). I haven't digged further as compiling with 2.4 is another issue.